app/main.py

```diff
@@ -1,2 +1,89 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table: list = [None] * self.capacity
+
+    def calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    @property
+    def current_max_size(self, threshold: float = 2 / 3) -> float:
```

Comment:
The `current_max_size` method is defined as a property but takes a parameter `threshold`. Properties should not take parameters. Consider converting this to a regular method or removing the parameter.